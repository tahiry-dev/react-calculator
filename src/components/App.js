import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import Calculate from '../logic/Calculate';

const App = () => {
  const [newState, setNewState] = useState({
    total: '',
    next: '',
    operation: null,
  });

  const handleClick = (buttonName) => {
    const currentState = Calculate(newState, buttonName);
    setNewState({ ...currentState });
  };

  const { total, next } = newState;

  return (
    <div className="container">
      <div className="calculator-container">
        <h2>Let&apos;s do some math!</h2>
        <div className="calculator">
          <Display total={total} next={next} />
          <ButtonPanel clickHandler={handleClick} />
        </div>
      </div>
    </div>
  );
};

export default App;
