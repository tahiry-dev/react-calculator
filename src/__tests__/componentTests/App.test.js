import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../components/App';

describe('App Component', () => {
    it('renders pages correctly', () => {
        const tree = renderer.create(<App />);
        expect(tree).toMatchSnapshot();
    });
});