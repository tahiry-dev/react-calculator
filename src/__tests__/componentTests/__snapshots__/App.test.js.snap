// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders App page correctly 1`] = `
<div
  className="container"
>
  <div
    className="calculator-container"
  >
    <h2>
      Let's do some math!
    </h2>
    <div
      className="calculator"
    >
      <div>
        <p
          className="result"
        >
          0
        </p>
      </div>
      <div>
        <div
          className="group1"
        >
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                AC
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                +/-
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                %
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              type="button"
            >
              <span>
                รท
              </span>
            </button>
          </div>
        </div>
        <div
          className="group2"
        >
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                7
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                8
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                9
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              type="button"
            >
              <span>
                x
              </span>
            </button>
          </div>
        </div>
        <div
          className="group3"
        >
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                4
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                5
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                6
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              type="button"
            >
              <span>
                -
              </span>
            </button>
          </div>
        </div>
        <div
          className="group4"
        >
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                1
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                2
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                3
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              type="button"
            >
              <span>
                +
              </span>
            </button>
          </div>
        </div>
        <div
          className="group5"
        >
          <div>
            <button
              className="zero"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                0
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                .
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              type="button"
            >
              <span>
                =
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App Component renders pages correctly 1`] = `
<div
  className="container"
>
  <div
    className="calculator-container"
  >
    <h2>
      Let's do some math!
    </h2>
    <div
      className="calculator"
    >
      <div>
        <p
          className="result"
        >
          0
        </p>
      </div>
      <div>
        <div
          className="group1"
        >
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                AC
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                +/-
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                %
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              type="button"
            >
              <span>
                รท
              </span>
            </button>
          </div>
        </div>
        <div
          className="group2"
        >
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                7
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                8
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                9
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              type="button"
            >
              <span>
                x
              </span>
            </button>
          </div>
        </div>
        <div
          className="group3"
        >
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                4
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                5
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                6
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              type="button"
            >
              <span>
                -
              </span>
            </button>
          </div>
        </div>
        <div
          className="group4"
        >
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                1
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                2
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                3
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              type="button"
            >
              <span>
                +
              </span>
            </button>
          </div>
        </div>
        <div
          className="group5"
        >
          <div>
            <button
              className="zero"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                0
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#e0e0e0",
                }
              }
              type="button"
            >
              <span>
                .
              </span>
            </button>
          </div>
          <div>
            <button
              onClick={[Function]}
              type="button"
            >
              <span>
                =
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
